### PR TITLE
rework category list and sorting

### PR DESCRIPTION
**Pull request by MaximilianKrambach**
A design and functionality overhaul for the category list.

Sorting is now an extra view, editing a category does not require an additional 'edit' button

Probably still a bit to do, but testable

### CODE REVIEW

**Comment by MaximilianKrambach** [comment]
An issue(?) I found was that 'getTitle' of motion is a bit nondescriptive. Having a category with motions labeled as  `Motion M001 ... Motion M099` doesn't really help if I want to reorder them.

**Comment by MaximilianKrambach** [comment]
> * Click on "Sort motion" reloads whole page -> use angular way to open new view!
I have no clue how the angular way is supposed to be here.

**Comment by MaximilianKrambach** [comment]
I had no success in properly catching the correct css of the sort-list (and shortening the motion title there), but I think this part is okay now.

The link being an reloading one is still open.

**Comment by MaximilianKrambach** [comment]
ping @emanuelschuetze

**Comment by emanuelschuetze** [comment]
Retested:
* Add more space on top for category list view (<div> with class="spacer-top-20"; see settings)
* Add space on bottom for new category new box (see motion blocks), use same width as list view box; use more space between prefix/name input fields in new

**Comment by MaximilianKrambach** [comment]
updated now @emanuelschuetze

**Comment by MaximilianKrambach** [comment]
Requesting a review from @tsiegleauq of the `sorting-list.component` because I changed the os-sorting-list behavior (in respect to updates and 'no data'), and I'm unsure if this may break anything in the other use cases (submitters list, speaker list) .

File: client/src/app/site/motions/components/category-list/category-list.component.scss [hunk 1]
```diff
@@ -1,33 +1,49 @@
...
     }
 }
 
-#updateForm {
-    margin-bottom: 20px;
+mat-expansion-panel {
+    max-width: 770px;
+    margin: auto;
+}
+.flex-spaced {
```
Comment (tsiegleauq) [inline client/src/app/site/motions/components/category-list/category-list.component.scss]: add one empty line

File: client/src/app/site/motions/components/category-sort/category-sort.component.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import { BaseViewComponent } from 'app/site/base/base-view';
+import { CategoryRepositoryService } from 'app/core/repositories/motions/category-repository.service';
+import { MatSnackBar } from '@angular/material';
+import { MotionRepositoryService } from 'app/core/repositories/motions/motion-repository.service';
+import { PromptService } from 'app/core/ui-services/prompt.service';
+import { SortingListComponent } from 'app/shared/components/sorting-list/sorting-list.component';
+import { ViewCategory } from '../../models/view-category';
+import { ViewMotion } from '../../models/view-motion';
+
+@Component({
```
Comment (tsiegleauq) [inline client/src/app/site/motions/components/category-sort/category-sort.component.ts]: doc missing

**Comment by tsiegleauq** [comment]
While testing: Ensure
- list of speakers
- motion sort submitters

are also still working and functional. Tested them briefly and could not find any defects.

**Comment by MaximilianKrambach** [comment]
@emanuelschuetze your comments have been adressed

**Pull request merged**